File: T6/codigos2021/T06_ComunicacionFragments/app/src/main/java/com/borja/t06_comunicacionfragments/fragments/FragmentUno.java
package com.borja.t06_comunicacionfragments.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.borja.t06_comunicacionfragments.R;

public class FragmentUno extends Fragment {

    private Button bComunicar;
    private EditText eTexto;
    private OnFragmentUnoListener listener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        listener = (OnFragmentUnoListener) context;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        return inflater.inflate(R.layout.fragment_uno_layout, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        bComunicar = getView().findViewById(R.id.boton_comunicar);
        eTexto = getView().findViewById(R.id.edit_fragment_uno);
        bComunicar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Toast.makeText(getContext(),"Mostrar TOAST desde fragment",Toast.LENGTH_SHORT).show();
                //listener.onFragmentUnoSelected();
                if (!eTexto.getText().toString().isEmpty()) {
                    listener.onFragmentTextSelected(eTexto.getText().toString());
                }
            }
        });
    }

    public interface OnFragmentUnoListener {
        void onFragmentUnoSelected();

        void onFragmentTextSelected(String texto);
    }

}

File: Kotlin/RECUPERACION/Naviegacion/app/src/main/java/com/example/navegacion/FragmentUno.kt
package com.example.navegacion

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.example.navegacion.databinding.FragmentUnoBinding

class FragmentUno : Fragment() {

    private lateinit var binding: FragmentUnoBinding
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        binding = FragmentUnoBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.botonF1.setOnClickListener {
            val bundle = Bundle();
            bundle.putString("nombre","Borja")
            findNavController().navigate(R.id.action_fragmentUno_to_fragmentDos,bundle)
        }
    }

}